crates/sui-move/src/unit_test.rs

```diff
@@ -125,9 +110,16 @@ pub fn run_move_unit_tests(
         sui_move_natives::all_natives(/* silent */ false),
         Some(initial_cost_schedule_for_unit_tests()),
         compute_coverage,
-        &mut std::io::sink(),
         &mut std::io::stdout(),
-    )
+    );
+    result.map(|(test_result, warning_diags)| {
+        if test_result == UnitTestResult::Success {
+            if let Some(diags) = warning_diags {
+                decorate_warnings(diags, None);
```

Comment:
Why only on success?

Reply:
It seems like when there are actual compilation errors, printing info linters stats and a place where to report linter issues is a bit noisy. Plus it would require even more changes in `run_move_unit_tests` to get the warnings in the error case  - certainly possible, but I am not sure necessary.

Reply:
Wait, we hide warnings on test success? What the fuck is that lol

Reply:
> Wait, we hide warnings on test success? What the fuck is that lol

No :-) We report warnings no matter what. We only show "decoration" (which is info about suppression and info about where to report linter issues) on test success. In this particular case, all diagnostics are already reported inside `move_cli::base::test::run_move_unit_tests`. The `warning_diags` returned from this function are only needed for decoration (need to count suppressions).